### PR TITLE
Add pagination for /apps route

### PR DESCRIPTION
**Pull request by maximemenager**
This PR adds a pagination for /apps route. The legacy limit is kept, but a custom limit can now be specified if needed

### CODE REVIEW

File: web/apps/apps.go [hunk 1]
```diff
@@ -301,7 +301,22 @@ func listWebappsHandler(c echo.Context) error {
 	if err := middlewares.AllowWholeType(c, permission.GET, consts.Apps); err != nil {
 		return err
 	}
-	docs, err := apps.ListWebapps(instance)
+
+	// Adding the startKey if it is given in the request
+	var startKey string
+	if sk := c.QueryParam("start_key"); sk != "" {
```
Comment (nono) [inline web/apps/apps.go]: Can you update the documentation?

File: model/app/konnector.go [hunk 2]
```diff
@@ -282,7 +282,6 @@ func GetKonnectorBySlugAndUpdate(db prefixer.Prefixer, slug string, copier appfs
 
 // ListKonnectors returns the list of installed konnectors applications.
 //
-// TODO: pagination
 func ListKonnectors(db prefixer.Prefixer) ([]Manifest, error) {
```
Comment (nono) [inline model/app/konnector.go]: Why did you keep this function?

File: model/app/konnector.go [hunk 3]
```diff
@@ -297,4 +296,32 @@ func ListKonnectors(db prefixer.Prefixer) ([]Manifest, error) {
...
+		Limit:    limit + 1, // Also get the following document for the next key
+		StartKey: startKey,
+	}
+	err := couchdb.GetAllDocs(db, consts.Konnectors, req, &docs)
+	if err != nil {
+		return nil, "", err
+	}
+
+	nextID := ""
+	if len(docs) > 0 && len(docs) == limit+1 { // There is still documents to fetch
```
Comment (nono) [inline model/app/konnector.go]: s/There is/There are/

File: model/app/konnector.go (hunk 2 above)
Comment (maximemenager) [reply model/app/konnector.go]: I was keeping it for the other uses (like intents), but with the same default limit (100), we can safely remove it without risking side-effects

File: web/apps/apps.go [hunk 4]
```diff
@@ -301,7 +301,22 @@ func listWebappsHandler(c echo.Context) error {
...
+		startKey = sk
+	}
```
Comment (nono) [inline web/apps/apps.go]: Why not just `startKey := c.QueryParam("start_key")`?

File: web/apps/apps.go [hunk 5]
```diff
@@ -320,15 +342,50 @@ func listKonnectorsHandler(c echo.Context) error {
...
+	// Generating links list for the next konnectors
+	links := &jsonapi.LinksList{}
+	if next != "" { // Do not generate the next URL if there are no next konnectors
+		links.Next = generateNextURL(c, limit, next)
+	}
+
+	return jsonapi.DataList(c, http.StatusOK, objs, links)
+}
+
+func generateNextURL(c echo.Context, limit int, nextID string) string {
```
Comment (nono) [inline web/apps/apps.go]: Can we move in this function the logic for?

```
	if next != "" { // Do not generate the next URL if there are no next konnectors
		links.Next = generateNextURL(c, limit, next)
	}
```

**Pull request merged**